File: Aufgabe1/Insertion Sort/sort.cpp
#include <vector>
void insertion_sort(std::vector<int>& v){
    int item;
    int temp;
    int k;
    int kleinstes;
    for(int i = 0; i < v.size(); i++){
        item = v.at(i);
        k = i+1;
        while(k < v.size()){
            if(v.at(k) < item){
                item = v.at(k);
                kleinstes = k;
            }
            k++;
        }
        if(item < v.at(i)){
            temp = v.at(i);
            v.at(i) = item;
            v.at(kleinstes) = temp;
        }
    }
}

File: Aufgabe1/Insertion Sort/sort.js

var x = [4,30,2,3,6,99,8,11,23];
console.log(Object.values(x));
y = insertion_sort(x);
console.log(Object.values(y));


function insertion_sort(x){
    for(i = 1; i < x.length; i++ ){
        k = x[i];
        j = i - 1;
        while(j >= 0 && x[j] > k){
            x[j+1] =  x[j];
            j--;
        }
        x[j+1] = k;
    }
    return x;
}